### PR TITLE
[libc++][z/OS] Correct a definition of __native_vector_size

### PR DESCRIPTION
**Pull request by zibi2**
Fix  `std/ranges/range.adaptors/range.lazy.split/general.pass.cpp` which started failing on z/OS after this [commit](https://github.com/llvm/llvm-project/commit/985c1a44f8d49e0af).

This test case is passing on other platforms such as AIX. This is because the `__ALTIVEC__` macro is defined and  `__mismatch` under `_LIBCPP_VECTORIZE_ALGORITHMS` guard is compiled out.  However, on z/OS  `_LIBCPP_VECTORIZE_ALGORITHMS`  is defined. Analyzing the algorithm of `__mismatch` shows that the culprit is the definition of `__native_vector_size` which was defined wrongly as 1. This PR corrects the definition of `__native_vector_size` and fixes the affected test.

Here is a subset of original test which fails on z/OS:

```
#include <ranges>
#include <cassert>
#include <vector>

template <class Char>
class BasicSmallString {
  std::vector<Char> buffer_{};

public:
  constexpr BasicSmallString(std::basic_string_view<Char> v)
    requires (std::same_as<Char, char> ||
              std::same_as<Char, wchar_t> ||
              std::same_as<Char, char8_t> ||
              std::same_as<Char, char16_t> ||
              std::same_as<Char, char32_t>)
    : buffer_(v.begin(), v.end())
  {}

  template <class I, class S>
  constexpr BasicSmallString(I b, const S& e) {
    for (; b != e; ++b) {
      buffer_.push_back(*b);
    }
  }

  template <std::ranges::range R>
  constexpr BasicSmallString(R&& from) : BasicSmallString(from.begin(), from.end()) {}

  friend constexpr bool operator==(const BasicSmallString& lhs, const BasicSmallString& rhs) {
    return lhs.buffer_ == rhs.buffer_;
  }
};

template <std::ranges::view View, std::ranges::range Expected>
constexpr bool is_equal(View& view, const Expected& expected) {
  using Char = std::ranges::range_value_t<std::ranges::range_value_t<View>>;
  using Str = BasicSmallString<Char>;

  auto actual_it = view.begin();
  auto expected_it = expected.begin();
  for (; actual_it != view.end() && expected_it != expected.end(); ++actual_it, ++expected_it) {
    if (Str(*actual_it) != Str(*expected_it))
      return false;
  }

  return actual_it == view.end() && expected_it == expected.end();
}

int main() {
  using namespace std::string_view_literals;
  auto overlapping_sep = "ab"sv;
  std::array expected = {"a"sv, "aa"sv, ""sv, "b"sv};
  std::ranges::lazy_split_view v("aabaaababb"sv, overlapping_sep);
  assert(is_equal(v, expected));
  return 0;
}
```

### CODE REVIEW

**Comment by mordante** [comment]
I see there are some CI failures with this patch.

**Comment by zibi2** [comment]
> I see there are some CI failures with this patch.
Yes, but the latest commit fixed them.

**Comment by zibi2** [comment]
The `generic-no-random_device` failure is unrelated. Restarted always fails with
`ninja: build stopped: interrupted by user.`

**Review by fanbo-meng — APPROVED** [review]
LGTM

**Comment by MaskRay** [comment]
Per consensus on https://discourse.llvm.org/t/hidden-emails-on-github-should-we-do-something-about-it/74223/32 , GitHub's ["private email address"](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address) feature should be avoided.

**Comment by philnik777** [comment]
Please wait for approval of the @llvm/reviewers-libcxx team next time.

**Comment by zibi2** [comment]
> Please wait for approval of the @llvm/reviewers-libcxx team next time.

Of course, I'll wait for approval next time.  Please let me know if NFC or trival commits like this one need special reviewers as well? Thank you

**Comment by philnik777** [comment]
Yes, generally any libc++ patch should be reviewed by a someone in the libc++ review group. In this case it's fine, but even trivial looking patches can have a significant ABI impact, which most people aren't aware of. I'd rather avoid even having the appearance of "anybody can approve" in libc++. If you have small patches like this feel free to ask on Discord whether someone can review it quickly.

**Comment by philnik777** [comment]
Just to be clear: Everybody is welcome (and encouraged) to review any libc++ patches. It's just about final approval.

**Comment by zibi2** [comment]
> Per consensus on https://discourse.llvm.org/t/hidden-emails-on-github-should-we-do-something-about-it/74223/32 , GitHub's ["private email address"](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address) feature should be avoided.

Thx, I was not aware of it.

**Pull request merged**
